server/issue_comment_handler.go

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package server
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/google/go-github/v32/github"
+	"github.com/mattermost/mattermost-server/v5/mlog"
+	"github.com/pkg/errors"
+)
+
+type issueCommentEvent struct {
+	Action     string                     `json:"action"`
+	Comment    *github.PullRequestComment `json:"comment"`
+	Issue      *github.Issue              `json:"issue"`
+	Repository *github.Repository         `json:"repository"`
+}
+
+func (s *Server) issueCommentEventHandler(w http.ResponseWriter, r *http.Request) {
+	ctx, cancel := context.WithTimeout(context.Background(), defaultRequestTimeout*time.Second)
+	defer cancel()
+
+	ev, err := issueCommentEventFromJSON(r.Body)
+	if err != nil {
+		mlog.Error("could not parse pr comment event", mlog.Err(err))
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	// We ignore comments from issues.
+	if !ev.Issue.IsPullRequest() {
+		return
+	}
+
+	pr, err := s.getPRFromIssueCommentEvent(ctx, ev)
+	if err != nil {
+		mlog.Error("Error getting PR from Comment", mlog.Err(err))
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	var commenter string
+	if ev.Comment != nil {
```

Comment:
Is this needed? We are already validation that the comment is not-nil in `issueCommentEventFromJSON`.